src/modele/communication/TransporteurMessage.java

```diff
@@ -61,10 +69,25 @@ public void receptionMessageDeSatellite(Message msg) {
         lock.lock();
 
         try {
-            /*
-             * (6.3.3) Insérer votre code ici
-             */
-        } finally {
+            if (msg instanceof Nack) {
+                // Si le message reçu est un Nack, on le met au début de la liste
+                messagesRecus.add(0, msg);
+            } else {
+                // Sinon, on calcule sa position selon le compteur
+                int position = 0;
+                for (int i = 0; i < messagesRecus.size(); i++) {
+                    Message currentMsg = messagesRecus.get(i);
+                    if (msg.getCompte() < currentMsg.getCompte()) {
+                        break;
+                    }
+                    position++;
+                }
+                // On ajoute le message à la position voulue
+                messagesRecus.add(position, msg);
+            }
+        } finally
+
+        {
             lock.unlock();
         }
```

Comment:
Simplification visuelle
```suggestion
        } finally {
            lock.unlock();
        }
```